math/src/elliptic_curve/short_weierstrass/curves/bls12_381/curve.rs

```diff
@@ -89,6 +96,25 @@ mod tests {
         assert_eq!(&g, &g);
     }
 
+    #[test]
+    fn g_operated_with_g_satifies_ec_equation() {
+        type FE = FieldElement<MontgomeryBackendPrimeField<BLS12381FieldModulus, 6>>;
+        let g = BLS12381Curve::generator();
+        let g2 = g.operate_with_self(FE::from(2).representative());
+
+        // get x and y from affine coordinates
+        let g2_affine = g2.to_affine();
+        let x = g2_affine.x();
+        let y = g2_affine.y();
+
+        // calculate both sides of BLS12-381 equation
+        let four = FieldElement::<MontgomeryBackendPrimeField<BLS12381FieldModulus, 6>>::from(4);
```

Comment:
And here `let four = FieldElement::from(4);`

Reply:
applied!